servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCodingHttpRequesterFilter.java

```diff
@@ -0,0 +1,138 @@
+        implements StreamingHttpClientFilterFactory, StreamingHttpConnectionFilterFactory,
+                   HttpExecutionStrategyInfluencer {
+
+    private final List<ContentCodec> supportedCodings;
+    @Nullable
+    private final CharSequence acceptedEncodingsHeader;
+
+    /**
+     * Enable support of the provided encodings for requests and responses.
+     * The order of the codecs provided, matters for the presentation of the header, and may affect selection priority
+     * on the receiving endpoint.
+     *
+     * @param supportedCodings the codecs this clients supports to encode/decode requests and responses accordingly
+     * and also used to advertise to the server.
+     */
+    public ContentCodingHttpRequesterFilter(final List<ContentCodec> supportedCodings) {
+        this.supportedCodings = supportedCodings;
+        this.acceptedEncodingsHeader = buildAcceptEncodingsHeader(supportedCodings);
+    }
+
+    @Override
+    public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+        return new StreamingHttpClientFilter(client) {
+            @Override
+            protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                            final HttpExecutionStrategy strategy,
+                                                            final StreamingHttpRequest request) {
+                return Single.defer(() -> codecTransformBidirectionalIfNeeded(delegate(), strategy, request));
+            }
+        };
+    }
+
+    @Override
+    public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnection connection) {
+        return new StreamingHttpConnectionFilter(connection) {
+            @Override
+            public Single<StreamingHttpResponse> request(final HttpExecutionStrategy strategy,
+                                                         final StreamingHttpRequest request) {
+                return Single.defer(() -> codecTransformBidirectionalIfNeeded(delegate(), strategy, request));
+            }
+        };
+    }
+
+    @Override
+    public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
+        // No influence since we do not block.
+        return strategy;
+    }
+
+    private Single<StreamingHttpResponse> codecTransformBidirectionalIfNeeded(final StreamingHttpRequester delegate,
+                                                                              final HttpExecutionStrategy strategy,
+                                                                              final StreamingHttpRequest request) {
+        final BufferAllocator alloc = delegate.executionContext().bufferAllocator();
+        setAcceptEncoding(request.headers(), acceptedEncodingsHeader);
+        encodePayloadContentIfAvailable(request, alloc);
+
+        return decodePayloadContentIfEncoded(delegate.request(strategy, request), alloc);
```

Comment:
The whole method should be wrapped with `defer` operator, similar to the service filter.

Reply:
The factory methods (`create`) do wrap it.